jerry-core/parser/regexp/re-compiler.c

```diff
@@ -180,7 +185,11 @@ re_insert_into_group (re_compiler_ctx_t *re_ctx_p, /**< RegExp compiler context
 
   qmin = re_ctx_p->current_token.qmin;
   qmax = re_ctx_p->current_token.qmax;
-  JERRY_ASSERT (qmin <= qmax);
+
+  if (qmin > qmax)
+  {
+    return ecma_raise_syntax_error (ECMA_ERR_MSG ("RegExp quantifier error: qmin > qmax."));
```

Comment:
qmin and qmax might be too internal concepts for a user. But the standard does talk about min and max, so it might be better to write min > max.

BTW, it might be useful to add a comment here referring to ES5.1 15.10.2.5 to note that the behaviour is prescribed there.

> If max is finite and less than min, then throw a SyntaxError exception.